PyInstaller/hooks/hook-sphinx.py

```diff
@@ -6,9 +6,80 @@
 #
 # The full license is in the file COPYING.txt, distributed with this software.
 #-----------------------------------------------------------------------------
-
-
+#
+# ********************************************
+# hook-sphinx.py - Pyinstaller hook for Sphinx
+# ********************************************
+# The following analysis applies to Sphinx v. 1.3.1, reported by ``pip show
+# sphinx``.
+#
+# From sphinx.application line 248::
+#
+#    __import__('sphinx.builders.' + mod, None, None, [cls]), cls)
+#
+# Therefore, we need all modules in ``sphinx.builders``.
 from PyInstaller.hooks.hookutils import collect_submodules, collect_data_files
 
-hiddenimports = collect_submodules('sphinx.ext')
-datas = collect_data_files('sphinx')
+hiddenimports = ( collect_submodules('sphinx.builders') +
+#
+# From sphinx.application line 429::
+#
+#    mod = __import__(extension, None, None, ['setup'])
+#
+# Per http://sphinx-doc.org/extensions.html#builtin-sphinx-extensions,
+# Sphinx extensions are all placed in ``sphinx.ext``. Include these.
+                  collect_submodules('sphinx.ext') +
+#
+# From sphinx.search line 228::
+#
+#    lang_class = getattr(__import__(module, None, None, [classname]),
+#                         classname)
+#
+# From sphinx. search line 119::
+#
+#    languages = {
+#        'da': 'sphinx.search.da.SearchDanish',
+#        'de': 'sphinx.search.de.SearchGerman',
+#        'en': SearchEnglish,
+#
+# So, we need all the languages in ``sphinx.search``.
+                  collect_submodules('sphinx.search') +
+#
+# From sphinx.websupport line 100::
+#
+#    mod = 'sphinx.websupport.search.' + mod
+#    SearchClass = getattr(__import__(mod, None, None, [cls]), cls)
+#
+# So, include modules under ``sphinx.websupport.search``.
+                  collect_submodules('sphinx.websupport.search') +
+#
+# From sphinx.util.inspect line 21::
+#
+#    inspect = __import__('inspect')
+#
+# And from sphinx.cmdline line 173::
+#
+#    locale = __import__('locale')  # due to submodule of the same name
+#
+# Add these two modules.
+                  ['inspect', 'locale',
+#
+# Plus, there are a HUGE number of imports from six that must be manually
+# listed. At some point, perhaps these can be auto-detected (see efforts
+# at https://github.com/pyinstaller/pyinstaller/pull/1231).
```

Comment:
OK, done. I'm very excited to see six support in both the Pyinstaller develop and python3 branches! Great work!